### PR TITLE
sudo: 1.9.15p4 -> 1.9.15p5

### PR DESCRIPTION
**Pull request by delroth**
## Description of changes

https://www.sudo.ws/releases/stable/#1.9.15p5 minor bug fixes.

Tested: nixosTests.sudo (x86_64 and aarch64), pkgsCross.aarch64-multiplatform.sudo.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by gepbird** [comment]
Result of `nixpkgs-review pr 277844` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><details><summary>mysql-workbench</summary><pre>
[5/788] Building CXX object tools/genobj/CMakeFiles/genobj.dir/genobj.cpp.o
FAILED: tools/genobj/CMakeFiles/genobj.dir/genobj.cpp.o 
/nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/g++ -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -DRAPIDJSON_HAS_STDSTRING -D__STDC_FORMAT_MACROS -I/build/>
In file included from /build/mysql-workbench-community-8.0.34-src/tools/genobj/genobj.cpp:32:
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2115:59: error: 'xmlNodePtr' has not been declared
 2115 |     static MetaClass *from_xml(const std::string &source, xmlNodePtr node);
      |                                                           ^~~~~~~~~~
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2153:19: error: 'xmlNodePtr' has not been declared
 2153 |     void load_xml(xmlNodePtr node);
      |                   ^~~~~~~~~~
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2154:30: error: 'xmlNodePtr' has not been declared
 2154 |     void load_attribute_list(xmlNodePtr node, const std::string &member = "");
      |                              ^~~~~~~~~~
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2534:5: error: 'xmlDocPtr' does not name a type
 2534 |     xmlDocPtr load_xml(const std::string &path);
      |     ^~~~~~~~~
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2535:27: error: 'xmlDocPtr' has not been declared
 2535 |     void get_xml_metainfo(xmlDocPtr doc, std::string &doctype_ret, std::string &version_ret);
      |                           ^~~~~~~~~
/build/mysql-workbench-community-8.0.34-src/library/grt/src/grt.h:2536:30: error: 'xmlDocPtr' has not been declared
 2536 |     ValueRef unserialize_xml(xmlDocPtr doc, const std::string &source_path);
      |                              ^~~~~~~~~
[6/788] Building CXX object ext/scintilla/gtk/CMakeFiles/wbscintilla.dir/__/src/CallTip.cxx.o
[7/788] Building CXX object ext/scintilla/gtk/CMakeFiles/wbscintilla.dir/__/src/AutoComplete.cxx.o
[8/788] Building CXX object ext/scintilla/gtk/CMakeFiles/wbscintilla.dir/__/src/CaseConvert.cxx.o
ninja: build stopped: subcommand failed.
</pre></details></li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>charliecloud</li>
    <li>check_zfs</li>
    <li>kitty</li>
    <li>kitty.kitten</li>
    <li>kitty.shell_integration</li>
    <li>kitty.terminfo</li>
    <li>lxqt.lxqt-sudo</li>
    <li>pmbootstrap</li>
    <li>pmbootstrap.dist</li>
    <li>qsudo</li>
    <li>sudo</li>
    <li>tests.testers.nixosTest-example</li>
    <li>tests.testers.runNixOSTest-example</li>
    <li>tests.trivial-builders.references</li>
  </ul>
</details>

Running `nix-build -A mysql-workbench` on commit 66bce6f2a026 passes, not sure why is it different for nixpkgs-review.

File: pkgs/tools/security/sudo/default.nix [hunk 1]
```diff
@@ -14,11 +14,11 @@
 
 stdenv.mkDerivation rec {
```
Comment (gepbird) [inline pkgs/tools/security/sudo/default.nix:15]: What about using the finalAttrs pattern?

**Review by gepbird — CHANGES_REQUESTED** [review]
Can you set "sudo" as the mainProgram? This package provides multiple binaries, but I believe sudo can be considered the main one.

File: pkgs/tools/security/sudo/default.nix (hunk 1 above)
Comment (delroth) [reply pkgs/tools/security/sudo/default.nix:15]: Sure, please send me a follow-up PR and I'll be happy to review it.

**Comment by delroth** [comment]
> Can you set "sudo" as the mainProgram? This package provides multiple binaries, but I believe sudo can be considered the main one.

We probably should, though it has extremely limited value since there is no reason you'd ever want to run binaries from this derivation directly without going through the NixOS wrapper.

Definitely not going to do it in this PR though unless you can think of any reason to block a version bump on unrelated changes.

**Pull request merged**